serial/serialwin32.py

```diff
@@ -285,11 +285,22 @@ def read(self, size=1):
                     ctypes.byref(self._overlapped_read))
                 if not read_ok and win32.GetLastError() not in (win32.ERROR_SUCCESS, win32.ERROR_IO_PENDING):
                     raise SerialException("ReadFile failed ({!r})".format(ctypes.WinError()))
-                result_ok = win32.GetOverlappedResult(
-                    self._port_handle,
-                    ctypes.byref(self._overlapped_read),
-                    ctypes.byref(rc),
-                    True)
+                # some dependencies rely on the broken timeout behavior, so avoiding fixing that for now
```

Comment:
Current timeout behavior effectively pads the value, so fixing it to be accurate here actually broke some downstream dependencies.  So, unfortunately, while this still fixes the KeyboardInterrupt problem, I had to revert the timeout fix for compatibility reasons.